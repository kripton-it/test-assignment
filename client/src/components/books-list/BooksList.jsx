import React from 'react';
import axios from 'axios';

import { BookItem } from '../index';

import { BooksListContainer } from './styles';

class BooksList extends React.Component {
  constructor() {
    super();
    this.state = {
      booksList: [],
      isLoading: true
    };
  }
  async componentDidMount() {
    const response = await axios.get("http://localhost:5000/books");
    this.setState({
      booksList: response.data.books,
      isLoading: false
    });
  }
  render() {
    const { booksList, isLoading } = this.state;
    if (isLoading) {
      return <h1>...loading</h1>;
    }
    return (
      <BooksListContainer>
        {booksList.map(book => <BookItem book={book} />)}
      </BooksListContainer>
    );
  }
}

export default BooksList;