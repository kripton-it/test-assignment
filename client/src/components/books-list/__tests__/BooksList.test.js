import React from "react";
import { shallow } from "enzyme";
import toJson from 'enzyme-to-json';

import BooksList from "../BooksList";

it("expect to render BooksList component", () => {
  const wrapper = shallow(<BooksList />);
  expect(toJson(wrapper)).toMatchSnapshot();
});