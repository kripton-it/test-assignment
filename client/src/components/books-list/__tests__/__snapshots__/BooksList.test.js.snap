// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render BooksList component 1`] = `<Spinner />`;
